source/distributed/tria.cc

```diff
@@ -1124,10 +1124,545 @@ namespace
 } // namespace
 
 
+
 namespace parallel
 {
   namespace distributed
   {
+    /* ------------------ class DataTransfer<dim,spacedim> ----------------- */
+
+
+    template <int dim, int spacedim>
+    Triangulation<dim, spacedim>::DataTransfer::DataTransfer(
+      MPI_Comm mpi_communicator)
+      : mpi_communicator(mpi_communicator)
+    {}
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::DataTransfer::pack_data(
+      const std::vector<quadrant_cell_relation_t> &quad_cell_relations,
+      const std::vector<typename CellAttachedData::pack_callback_t>
+        &pack_callbacks)
+    {
+      Assert(src_data_fix.capacity() == 0,
```

Comment:
that seems like the wrong check. `capacity()` returns how many elements a vector *can* store, not how many it *actually* stores. Did you mean to use `size()`?

Reply:
Either should work in this case.

I wanted to make sure that the memory is completely flushed with the <tt>DataTransfer::clear</tt> function beforehand. But since no other function changes the size or capacity of that member, we can also take <tt>size()</tt> here. I'll change it, since it is more intuitive.

Reply:
My impression is that what capacity a vector has at any given time is an implementation detail that we can not affect. For example, https://en.cppreference.com/w/cpp/container/vector/clear states that this leaves the capacity unchanged -- though, interestingly, http://www.cplusplus.com/reference/vector/vector/clear/ has different wording. Either way, `capacity()` should not be trusted except as an *upper bound*.